File: 25-03-18/c++/01-Xsquare-and-two-Arrays.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    int n, q;
    cin >> n >> q;
    int ar1[n+1], ar2[n+1];
    ar1[0] = 0;
    ar2[0] = 0;
    for(int i = 1; i <= n; i++)
        cin >> ar1[i];
    for(int i = 1; i <= n; i++)
        cin >> ar2[i];
    
    long long dp1[n+1] = {0}, dp2[n+1] = {0};
    
    dp1[1] = ar1[1];
    dp2[1] = ar2[1];
    
    for(int i = 2; i <= n; i++)
    {
        if(i % 2 == 0)
        {
            dp1[i] = dp1[i-1] + ar2[i];
            dp2[i] = dp2[i-1] + ar1[i];
        }
        else
        {
            dp1[i] = dp1[i-1] + ar1[i];
            dp2[i] = dp2[i-1] + ar2[i];
        }
    }
    
    while(q--)
    {
        int a, x, y;
        cin >> a >> x >> y;
        
        if(a == 1)
        {
            if(x % 2 != 0 )
            {
                cout << (dp1[y] - dp1[x-1]) << endl;
            }
            else
            {
                cout << (dp2[y] - dp2[x-1]) << endl;
            }
        }
        else
        {
            if(x % 2 == 0 )
            {
                cout << (dp1[y] - dp1[x-1]) << endl;
            }
            else
            {
                cout << (dp2[y] - dp2[x-1]) << endl;
            }
        }
    }
    
}

File: 25-03-18/java/01-Xsquare-and-two-Arrays.java

import java.util.*;


class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int q = input.nextInt();
        
        int ar1[] = new int[n+1];
        int ar2[] = new int[n+1];
        long dp1[] = new long[n+1];
        long dp2[] = new long[n+1];
        
        for(int i = 1; i <= n; i++)
            ar1[i] = input.nextInt();
        for(int i = 1; i <= n; i++)
            ar2[i] = input.nextInt();
        
        for(int i = 1; i <= n; i++)
        {
            if(i%2==0)
            {
                dp1[i] = dp1[i-1] + ar2[i];
                dp2[i] = dp2[i-1] + ar1[i];
            }
            else
            {
                dp1[i] = dp1[i-1] + ar1[i];
                dp2[i] = dp2[i-1] + ar2[i];
            }
        }
        
        while(q-- > 0)
        {
            int a, x, y;
            a = input.nextInt();
            x = input.nextInt();
            y = input.nextInt();
            if(a == 1)
            {
                if(x % 2 != 0)
                {
                    System.out.println(dp1[y] - dp1[x-1]);
                }
                else
                {
                    System.out.println(dp2[y] - dp2[x-1]);
                }
            }
            else
            {
                if(x % 2 == 0)
                {
                    System.out.println(dp1[y] - dp1[x-1]);
                }
                else
                {
                    System.out.println(dp2[y] - dp2[x-1]);
                }
            }
        }
        
        
    }
}
